### PR TITLE
chore: convert mocha tests and test helpers to esmodules

### PR DESCRIPTION
**Pull request by BeksOmega**
## The basics

- [X] I branched from develop
- [X] My pull request is against develop
- [X] My code follows the [style guide](https://developers.google.com/blockly/guides/modify/web/style-guide)
- [X] I ran `npm run format` and `npm run lint`

### Resolves

N/A

### Proposed Changes

Converts the mcoah tests to esmodules.

#### Behavior Before Change

No change in behavior.

#### Behavior After Change

No change in behavior.

### Reason for Changes

If we want to move to webpack, it is a good idea to convert the tests to esmodules. If we want to run the tests in node rather than webdriver, it is also a good idea to convert the tests to esmodules.

### Test Coverage

This is tests silly!

### Documentation

N/A

### Additional Information

There are still some `goog.declareModuleId` statements so that the `index.html` file can properly import these. And there are still `goog.require` statements because the blocks and generators have not yet been converted.

### CODE REVIEW

File: tests/mocha/blocks/lists_test.js [hunk 1]
```diff
@@ -4,12 +4,12 @@
  * SPDX-License-Identifier: Apache-2.0
  */
 
-goog.module('Blockly.test.lists');
+goog.declareModuleId('Blockly.test.lists');
 
 const {runSerializationTestSuite} = goog.require('Blockly.test.helpers.serialization');
```
Comment (rachel-fenichel) [inline tests/mocha/blocks/lists_test.js:9]: Why are the serialization helpers and ConnectionType requires not converted?

File: tests/mocha/blocks/logic_ternary_test.js [hunk 2]
```diff
@@ -4,11 +4,11 @@
  * SPDX-License-Identifier: Apache-2.0
  */
 
-goog.module('Blockly.test.logicTernary');
+goog.declareModuleId('Blockly.test.logicTernary');
 
 const eventUtils = goog.require('Blockly.Events.utils');
```
Comment (rachel-fenichel) [inline tests/mocha/blocks/logic_ternary_test.js:9]: Event utils can be converted

File: tests/mocha/blocks/procedures_test.js [hunk 3]
```diff
@@ -4,13 +4,13 @@
  * SPDX-License-Identifier: Apache-2.0
  */
 
-goog.module('Blockly.test.procedures');
+goog.declareModuleId('Blockly.test.procedures');
 
 goog.require('Blockly');
```
Comment (rachel-fenichel) [inline tests/mocha/blocks/procedures_test.js:9]: In one of the other files you converted the require for Blockly. Why not here?

**Comment by BeksOmega** [comment]
I think the issue was that files in `mocha/blocks` were moved there in develop, but not ts/migration (which this was previously branched off of) so there were some errors with the rebase. Fixed.

**Review by rachel-fenichel — APPROVED** [review]

**Pull request merged**